ibm/resource_ibm_cis_rate_limit.go

```diff
@@ -338,220 +338,229 @@ func resourceIBMCISRateLimitValidator() *ResourceValidator {
-	if err != nil {
-		return err
-	}
-	recordID, zoneID, cisID, _ := convertTfToCisThreeVar(d.Id())
+	cisClient, err := meta.(ClientSession).CisRLClientSession()
 	if err != nil {
 		return err
 	}
-	updateRecord := v1.RateLimitRecord{}
-	if d.HasChange("disabled") || d.HasChange(cisRLThreshold) || d.HasChange(cisRLPeriod) || d.HasChange(cisRLDescription) || d.HasChange("action") || d.HasChange("match") || d.HasChange("correlate") || d.HasChange("bypass") {
 
-		updateRecord.Threshold = d.Get(cisRLThreshold).(int)
-		updateRecord.Period = d.Get(cisRLPeriod).(int)
+	cisID := d.Get("cis_id").(string)
+	zoneID := d.Get("domain_id").(string)
+	cisClient.Crn = core.StringPtr(cisID)
+	cisClient.ZoneIdentifier = core.StringPtr(zoneID)
+	ruleID := d.Get("rule_id").(string)
+
+	opt := cisClient.NewUpdateRateLimitOptions(ruleID)
+	if d.HasChange("disabled") ||
+		d.HasChange(cisRLThreshold) ||
+		d.HasChange(cisRLPeriod) ||
+		d.HasChange(cisRLDescription) ||
+		d.HasChange("action") ||
+		d.HasChange("match") ||
+		d.HasChange("correlate") ||
+		d.HasChange("bypass") {
+
+		opt.SetThreshold(int64(d.Get(cisRLThreshold).(int)))
+		opt.SetPeriod(int64(d.Get(cisRLPeriod).(int)))
 
 		if description, ok := d.GetOk(cisRLDescription); ok {
-			updateRecord.Description = description.(string)
+			opt.SetDescription(description.(string))
 		}
 
 		if disabled, ok := d.GetOk("disabled"); ok {
-			updateRecord.Disabled = disabled.(bool)
+			opt.SetDisabled(disabled.(bool))
 		}
 
 		action, err := expandRateLimitAction(d)
 		if err != nil {
 			return fmt.Errorf("Error in getting action from expandRateLimitAction %s", err)
 		}
-		updateRecord.Action = action
+		opt.SetAction(action)
 
 		match, err := expandRateLimitMatch(d)
 		if err != nil {
 			return fmt.Errorf("Error in getting match from expandRateLimitMatch %s", err)
 		}
-		updateRecord.Match = match
+		opt.SetMatch(match)
 
 		correlate, err := expandRateLimitCorrelate(d)
 		if err != nil {
 			return fmt.Errorf("Error in getting correlate from expandRateLimitCorrelate %s", err)
 		}
-		updateRecord.Correlate = &correlate
+		opt.SetCorrelate(correlate)
 
 		byPass, err := expandRateLimitBypass(d)
 		if err != nil {
 			return fmt.Errorf("Error in getting bypass from expandRateLimitBypass %s", err)
 		}
-		updateRecord.Bypass = byPass
-
-	}
-	_, err = cisClient.RateLimit().UpdateRateLimit(cisID, zoneID, recordID, updateRecord)
-	if err != nil {
-		return fmt.Errorf("Failed to update RateLimit: %v", err)
+		opt.SetBypass(byPass)
+		_, resp, err := cisClient.UpdateRateLimit(opt)
+		if err != nil {
+			return fmt.Errorf("Failed to update RateLimit: %v", resp)
+		}
 	}
 	return resourceIBMCISRateLimitRead(d, meta)
 }
 
 func resourceIBMCISRateLimitDelete(d *schema.ResourceData, meta interface{}) error {
-	cisClient, err := meta.(ClientSession).CisAPI()
+	cisClient, err := meta.(ClientSession).CisRLClientSession()
 	if err != nil {
 		return err
 	}
-	recordID, zoneID, cisID, _ := convertTfToCisThreeVar(d.Id())
```

Comment:
donot do get in delete method,
get it from the id

Reply:
done